app/scripts/controllers/decrypt-message.ts

```diff
@@ -216,7 +236,12 @@ export default class DecryptMessageController extends BaseControllerV2<
         await this._decryptMessageManager.approveMessage(messageParams);
 
       cleanMessageParams.data = this._parseMessageData(cleanMessageParams.data);
-      const rawMessage = await this._keyringController.decryptMessage(
+      if (!isEIP1024EncryptedMessage(cleanMessageParams)) {
+        throw new Error('Invalid encrypted data.');
```

Comment:
Could you add a test case for each of these new validation errors? It looks like they aren't yet captured